pkg/config/config.go

```diff
@@ -96,7 +96,24 @@ func NewConfig() (*Config, error) {
 	if err != nil {
 		return nil, err
 	}
-	return c, err
+
+	return c, nil
+}
+
+func NewConfig() (*Config, error) {
+	c, err := newConfigInternal()
+	if err != nil {
+		return nil, err
+	}
+	missingKeys := GetMissingRequiredKeys(c.values, "mapstructure", "squash", "required")
```

Comment:
consider using "verify" or other tag names that will hold "required" as value, later we can add more constraints.